chunky/src/java/se/llbit/chunky/world/Chunk.java

```diff
@@ -161,14 +161,14 @@ public synchronized boolean loadChunk(ChunkData chunkData, int yMin, int yMax) {
 
     surfaceTimestamp = dataTimestamp;
     version = chunkVersion(data);
-    loadSurface(data, chunkData, yMin, yMax);
+    loadSurface(data, blockPalette, biomePalette, chunkData, yMin, yMax);
     biomesTimestamp = dataTimestamp;
 
     world.chunkUpdated(position);
     return true;
   }
 
-  private void loadSurface(@NotNull Tag data, ChunkData chunkData, int yMin, int yMax) {
+  private void loadSurface(@NotNull Tag data, BlockPalette blockPalette, BiomePalette biomePalette, ChunkData chunkData, int yMin, int yMax) {
```

Comment:
Should we add a `ChunkDataContainer` that contains the chunk data, biome palette and block palette?